windpowerlib/wind_turbine.py

```diff
@@ -172,6 +172,18 @@ def __init__(self, hub_height, nominal_power=None, path='oedb',
                    " Check if the turbine type {0} is in your database file"
                    " or if you passed a valid curve.")
             warnings.warn(msg.format(turbine_type), WindpowerlibUserWarning)
+        else:
+            # power (coefficient) curve to pd.DataFrame in case of being dict
+            if isinstance(self.power_curve, dict):
+                self.power_curve = pd.DataFrame(self.power_curve)
+            if isinstance(self.power_coefficient_curve, dict):
+                self.power_coefficient_curve = pd.DataFrame(
+                    self.power_coefficient_curve)
+            # sort power (coefficient) curve by wind speed
+            if isinstance(self.power_curve, pd.DataFrame):
```

Comment:
First I thought this check if the power curve was a dataframe was unnecessary but then I realized that we didn't check this at all yet in case of a power curve directly provided by the user. I would therefore suggest to leave this check and add an else which would throw an error in case the type of the provided power curve is not dict or dataframe. What do you think @SabineHaas?